pkg/process/net/common.go

```diff
@@ -159,6 +161,36 @@ func (r *RemoteSysProbeUtil) GetConnections(clientID string) (*model.Connections
 	return conns, nil
 }
 
+// GetConnectionsGRPC returns a set of active network connections, retrieved from the system probe grpc server
+func (r *RemoteSysProbeUtil) GetConnectionsGRPC(clientID, unixPath string) (*model.Connections, error) {
+	// Create a context with a timeout of 10 seconds
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	conn, err := grpc.Dial("unix://"+unixPath, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	client := connectionserver.NewSystemProbeClient(conn)
+
+	response, err := client.GetConnections(ctx, &connectionserver.GetConnectionsRequest{ClientID: clientID})
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := response.Recv()
+	if err != nil {
+		return nil, err
+	}
+
+	conns, err := netEncoding.GetUnmarshaler("application/protobuf").Unmarshal(res.Data)
```

Comment:
i think we can simply go with
```suggestion
	return netEncoding.GetUnmarshaler("application/protobuf").Unmarshal(res.Data)
```